File: core/src/main/java/com/fahamutech/kcore/admin/database/connector/HomeDataSource.java
package com.fahamutech.kcore.admin.database.connector;

import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
import androidx.recyclerview.widget.RecyclerView;

public interface HomeDataSource {

    Object getCategory(RecyclerView recyclerView, SwipeRefreshLayout swipeRefreshLayout);

    Object getTestimony(RecyclerView recyclerView, SwipeRefreshLayout swipeRefreshLayout);

}

File: core/src/main/java/com/fahamutech/kcore/admin/database/connector/HomeDataSource.kt
package com.fahamutech.kcore.admin.database.connector

import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import androidx.recyclerview.widget.RecyclerView

interface HomeDataSource {
    fun getCategory(recyclerView: RecyclerView?, swipeRefreshLayout: SwipeRefreshLayout?): Any?
    fun getTestimony(recyclerView: RecyclerView?, swipeRefreshLayout: SwipeRefreshLayout?): Any?
}